File: leetcode/Python/4sum-ii.py
class Solution:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        um, sol, sz = defaultdict(int), 0, len(nums1)
        for i in range(sz):
            for j in range(sz):
                um[nums1[i]+nums2[j]]+=1
        for i in range(sz):
            for j in range(sz):
                val = nums3[i]+nums4[j]
                sol+=um[-val]
        return sol

File: leetcode/C++/4sum-ii.cpp
class Solution {
public:
    int fourSumCount(vector<int>& A, vector<int>& B, vector<int>& C, vector<int>& D) {
        int answer=0, len = A.size();
        long long val=0;
        unordered_map<long long,int> m;
        for(int i=0; i<len; ++i){
            for(int j=0; j<len; ++j){
                ++m[A[i]+B[j]];
            }
        }
        for(int i=0; i<len; ++i){
            for(int j=0; j<len; ++j){
                val=C[i]+D[j];
                answer+=m[-val];
            }
        }
        return answer;
    }
};